src/migrate.js

```diff
@@ -40,20 +41,22 @@ if ( jQuery.migrateTrace === undefined ) {
 // Forget any warnings we've already given; public
 jQuery.migrateReset = function() {
 	warnedAbout = {};
-	jQuery.migrateWarnings.length = 0;
+	jQuery.migrateWarnings.length = jQuery.migrateWarnings.asErrors.length = 0;
 };
 
 function migrateWarn( msg ) {
 	var console = window.console;
 	if ( !warnedAbout[ msg ] ) {
 		warnedAbout[ msg ] = true;
 		jQuery.migrateWarnings.push( msg );
+		jQuery.migrateWarnings.asErrors.push( new Error( msg ) );
 		if ( console && console.warn && !jQuery.migrateMute ) {
 			console.warn( "JQMIGRATE: " + msg );
 			if ( jQuery.migrateTrace && console.trace ) {
 				console.trace();
 			}
 		}
+		jQuery( window ).trigger( "migrateWarning" );
```

Comment:
I have to say I hadn't thought of intercepting the console. Obviously that would require unmuting jquery-migrate, and adding the wrapper in that intercept would be another uninformative line in the stack trace, but that's certainly no dealbreaker. Would scraping for /^JQMIGRATE:/ be safe both now and in the future for consumers who have other cruft in their console they aren't interested in recording?

I understand your hesitation about looping back through .trigger, although it would certainly be possible if you _were_ duckpunching trigger to call the original directly in that one case (or presumably if you were raising warnings, there's a decent chance it'd be because some other way would be preferred).